### PR TITLE
Use Array instead of jnp.array in jax.random

### PR DESCRIPTION
**Pull request by NeilGirdhar**
This satisfies MyPy for the functions:
* jax.random.categorical,
* jax.random.shuffle, and
* jax.random.permutation.

### CODE REVIEW

**Comment by NeilGirdhar** [comment]
@jakevdp Sorry for the duplicate; github won't let me reopen https://github.com/google/jax/pull/6709.

**Review by jakevdp — APPROVED** [review]
Looks good - thanks!
